zebra/kernel_netlink.c

```diff
@@ -1062,20 +1072,8 @@ int netlink_request(struct nlsock *nl, void *req)
 	n->nlmsg_pid = nl->snl.nl_pid;
 	n->nlmsg_seq = ++nl->seq;
 
-	memset(&snl, 0, sizeof(snl));
-	snl.nl_family = AF_NETLINK;
-
-	/* Raise capabilities and send message, then lower capabilities. */
-	frr_with_privs(&zserv_privs) {
-		ret = sendto(nl->sock, req, n->nlmsg_len, 0,
-			     (struct sockaddr *)&snl, sizeof(snl));
-	}
-
-	if (ret < 0) {
-		zlog_err("%s sendto failed: %s", nl->name,
-			 safe_strerror(errno));
+	if (netlink_send_msg(nl, req, n->nlmsg_len) < 0)
```

Comment:
Nit: `netlink_send_msg` returns the value of `sendmsg` so it can only be `-1` on error.